Document/0x05h-Testing-Platform-Interaction.md

```diff
@@ -297,42 +297,34 @@ When analyzing permissions, you should investigate the concrete use case scenari
 
 ### Dynamic Analysis
 
-Permissions for installed applications can be retrieved with Drozer. The following extract demonstrates how to examine the permissions used by an application and the custom permissions defined by the app:
+Permissions for installed applications can be retrieved with `adb`. The following extract demonstrates how to examine the permissions used by an application.
 
 ```bash
-dz> run app.package.info -a com.android.mms.service
-Package: com.android.mms.service
-  Application Label: MmsService
-  Process Name: com.android.phone
-  Version: 6.0.1
-  Data Directory: /data/user/0/com.android.mms.service
-  APK Path: /system/priv-app/MmsService/MmsService.apk
-  UID: 1001
-  GID: [2001, 3002, 3003, 3001]
-  Shared Libraries: null
-  Shared User ID: android.uid.phone
-  Uses Permissions:
-  - android.permission.RECEIVE_BOOT_COMPLETED
-  - android.permission.READ_SMS
-  - android.permission.WRITE_SMS
-  - android.permission.BROADCAST_WAP_PUSH
-  - android.permission.BIND_CARRIER_SERVICES
-  - android.permission.BIND_CARRIER_MESSAGING_SERVICE
-  - android.permission.INTERACT_ACROSS_USERS
-  Defines Permissions:
-  - None
+$ adb shell dumpsys package com.google.android.youtube
+...
+declared permissions:
+  com.google.android.youtube.permission.C2D_MESSAGE: prot=signature, INSTALLED
+requested permissions:
+  android.permission.INTERNET
+  android.permission.ACCESS_NETWORK_STATE
+install permissions:
+  com.google.android.c2dm.permission.RECEIVE: granted=true
+  android.permission.USE_CREDENTIALS: granted=true
+  com.google.android.providers.gsf.permission.READ_GSERVICES: granted=true  
+...
 ```
 
-When Android applications expose IPC components to other applications, they can define permissions to control which applications can access the components. For communication with a component protected by a `normal` or `dangerous` permission, Drozer can be rebuilt so that it includes the required permission:
+From the command above, the following bloc are interesting, the information below are based on this [blog post](https://www.programmersought.com/article/47364329867/ "See Android permissions from dumpsys package"):
```

Comment:
```suggestion
From the command above, the following bloc are interesting, the information below are based on this [blog post](https://www.programmersought.com/article/47364329867/ "See Android permissions from dumpsys package"):
```
I think bloc is a typo, I'm not sure what it should mean.

Reply:
Suggestion commited.

Reply:
Typo fixed